cmake/kokkos_enable_options.cmake

```
@@ -107,6 +107,16 @@ kokkos_enable_option(
 )
 mark_as_advanced(Kokkos_ENABLE_IMPL_VIEW_OF_VIEWS_DESTRUCTOR_PRECONDITION_VIOLATION_WORKAROUND)
 
+kokkos_enable_option(EXPERIMENTAL_CXX20_MODULE OFF "Whether to build a kokkos C++20 module")
```

Not sure about the description.  Should we say "enable C++20 module support"?

---

As user, I wouldn't know what "enable C++20 module support means". I might assume that means that modules are used internally but the point rather is that we are exporting a module interface.
```suggestion
kokkos_enable_option(EXPERIMENTAL_CXX20_MODULES OFF "Whether to export C++20 modules for Kokkos.")
```